File: archive-romhandbook-ui/lib/types/Buff.ts
export type Buff = {

    id: string

    name: string

    detail_url: string

    image: string | null

    description: string | null

    raw_json: string | null
}

export type BuffDetail = {

    id: string

    name: string

    detail_url: string

    image: string | null

    description: string | null

    raw_json: string | null

    raw_html: string | null
}

export type ApiResponse<T> = {

    success: boolean

    data: T

    meta: unknown
}

export type PaginatedApiResponse<T> = {

    success: boolean

    data: T[]

    meta: {
        page: number
        limit: number
        total: number
        has_next: boolean
    }
}

File: backend-api/internal/models/buff.go
package models

type Buff struct {
	ID string `json:"id"`

	Name string `json:"name"`

	DetailURL string `json:"detail_url"`

	Image *string `json:"image"`

	Description *string `json:"description"`

	RawJSON *string `json:"raw_json,omitempty"`
}

type BuffDetail struct {
	ID string `json:"id"`

	Name string `json:"name"`

	DetailURL string `json:"detail_url"`

	Image *string `json:"image"`

	Description *string `json:"description"`

	RawJSON *string `json:"raw_json"`

	RawHTML *string `json:"raw_html"`
}
